Source/WebCore/editing/FrameSelection.h

```diff
@@ -111,7 +112,7 @@ class DragCaretController : private CaretBase {
     VisiblePosition m_position;
 };
 
-class FrameSelection final : private CaretBase, public CaretAnimationClient {
+class FrameSelection final : private CaretBase, public CaretAnimationClient, public CanMakeCheckedPtr {
```

Comment:
Originally we thought of these sub-objects of `Frame` as not having their own lifetime. Thus we’d really just want to forward the checking to `Frame` instead of having separate checking of the sub-objects. Same for strong references and weak references. I think an idiom like that would be more efficient and just as safe.

Reply:
FrameSelection is a bit complicated. It is normally owned by a Document but not necessarily. We have the following code in accessibility:
```
    FrameSelection selection;
    selection.setSelection(position);
    selection.modify(FrameSelection::Alteration::Extend, SelectionDirection::Right, TextGranularity::LineBoundary);
    return selection.selection();
```

Reply:
OK we have to live with this for now, but there should be a plain data holder for the local variable without the word frame in its name.